tests/testerina-integration-test/src/test/resources/single-file-tests/grouping/before-groups-after-groups-test.bal

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Test file to test the test execution behavior when @BeforeGroups and @AfterGroups functions
+# are defined. The expected behavior is that, each @BeforeGroups function will execute before the first
+# test of that group is executed and each @AfterGroups function will execute after the
+# last test of that group is executed.
+
+import ballerina/io;
+import ballerina/test;
+
+string a = "";
+
+@test:BeforeGroups { value : ["g1"] }
```

Comment:
Shall we add "beforeSuite" function as well to the test?

Reply:
Added in 019b9c53e2d3bafcc56689504c433876f1e1770a